### DISCUSSION TITLE
Do I really need authExchange or I can make it simpler?

### DISCUSSION THREAD

**Discussion by hexwit** [urql-graphql/urql#1815] [Q&A]
Hi! 

Situation is following: I have authentication based on cookie + csrf token (no JWT here). 
CSRF token one per session. So here is nothing to refresh.

Technically all I need is add auth to each request, and detect "access denied" error to prevent further requests 
and redirect to the login page. 

According to the description of authExchange work in https://github.com/FormidableLabs/urql/pull/939, 
exchange implies having refreshable token. That is why getAuth called multiple times, etc etc. 
For my scenario it is a bit overcomplicated. 

Is there any other way to achieve my goal without authExchange?

Thanks!

**Comment by kitten** [comment]
I mean, you can totally do this without by using `fetchOptions` to pass your token and the `onError` exchange to detect auth errors. However, the same can also be done with `authExchange`. Nothing basically stops you from redirecting in `getAuth` after its first call, so that's basically what I'd advise, since that's the quickest way to implement this and still allows you to utilise `willAuthError` to predictively trigger `getAuth`.

What's important to understand here is that `getAuthg` being called multiple times isn't overcomplicating things, technically 😅  It's just called again to refresh the auth state. Whether you then actually refresh it or redirect to a different page shouldn't matter much
